win32ss/user/ntuser/winpos.c

```diff
@@ -2256,15 +2256,19 @@ co_WinPosSetWindowPos(
             //ERR("WPSWP : set active window\n");
             if (!(Window->state & WNDS_BEINGACTIVATED)) // Inside SAW?
             {
+               // Set state flag to prevent recursions.
+               BOOL WasBeingActivated = (Window->state & WNDS_BEINGACTIVATED) != 0;
```

Comment:
If the `State` value can change, should a thread-safe "function" be used to test/set this bit (everywhere)?

Reply:
win32k is protected with a giant lock, so if two or more GUI threads invoke window stuff, all except one should be blocked; so one thread only should run this code and set the internal bit there.

Reply:
Thanks for the explanation.
Then, isn't [Try fix](https://github.com/reactos/reactos/pull/7555/commits/f3e1fb080aa7f7f399d8872a7b947eb100f1e0d3) commit useless?